packages/gatsby/src/utils/worker/types.ts

```diff
@@ -0,0 +1,24 @@
+import type Worker from "jest-worker"
+
+type WrapReturnOfAFunctionInAPromise<
+  FunctionThatDoesNotReturnAPromise extends (...args: Array<any>) => any
+> = (
+  ...a: Parameters<FunctionThatDoesNotReturnAPromise>
+) => Promise<ReturnType<FunctionThatDoesNotReturnAPromise>>
+
+// jest-worker will make sync function async, so to keep proper types we need to adjust types so all functions
+// on worker pool are async
+type EnsureFunctionReturnsAPromise<MaybeFunction> = MaybeFunction extends (
+  ...args: Array<any>
+) => Promise<any>
+  ? MaybeFunction
+  : MaybeFunction extends (...args: Array<any>) => any
+  ? WrapReturnOfAFunctionInAPromise<MaybeFunction>
+  : never
```

Comment:
`never` to disallow non-functions if export is not a function

Reply:
This is complex XD isn't there another way to accomplish the same or not really?

Reply:
My TS-foo is not strong, so not sure about simpler way (at least to have those types automatically) - there might, I just don't know of any

I initially just used exported types from `child` module as they are (the way `jest` itself does it - https://github.com/facebook/jest/blob/5f4dd187d89070d07617444186684c20d9213031/packages/jest-runner/src/index.ts#L38-L40), but if you have `sync` method in child - using it on `workerPool` would return promise anyway, so types were not always correct.

I guess simple approach would be to define types of "exposed functions" manually, but then it's possibly error-prone and easy to get out of sync.